reasoner_pydantic/results.py

```diff
@@ -174,35 +175,53 @@ def combine_analyses_by_resource_id(self):
 class Results(BaseModel):
     """Results."""
 
-    __root__: Optional[HashableSet[Result]]
+    __root__: Optional[HashableSequence[Result]]
 
     class Config:
         title = "results"
         extra = "allow"
 
+    def append(self, result):
+        self.__root__.append(result)
+
     def add(self, result):
         results = self.__root__
-        if result in results:
+        try:
             # this is slow for larger results
-            for original_result in results:
-                if result == original_result:
-                    original_result.update(result)
-                    return
-        else:
-            results.add(result)
+            results[results.index(result)].update(result)
+        except ValueError:
+            results.append(result)
 
     def __len__(self):
         return len(self.__root__)
 
     def __iter__(self):
         return self.__root__.__iter__()
+    
+    def __contains__(self, v):
+        return self.__root__.__contains__(v)
+    
+    def __getitem__(self, i):
+        return self.__root__.__getitem__(i)
 
     def update(self, other):
-        for result in other.__root__:
-            self.add(result)
+        results = parse_obj_as(HashableMapping, {})
+        for result in other:
+            result = Result.parse_obj(result)
+            result_hash = hash(result)
+            if result_hash in results:
+                results[result_hash].update(result)
+            else:
+                results[hash(result)] = result
 
     def parse_obj(obj):
-        results = parse_obj_as(Results, obj)
+        parse_obj_as(Results, obj)
```

Comment:
Is this line supposed to be here?

Reply:
Yes, this is what does the actual pydantic validation.